-1

Tested at: 3b3c1c20403e4a6379e3757344c782c55b2466ea

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18166/18857/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>In file included from /cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/boost/1.63.0-mlhled/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:63:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/boost/1.63.0-mlhled/include/boost/lexical_cast/detail/inf_nan.hpp:34:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/boost/1.63.0-mlhled/include/boost/math/special_functions/sign.hpp:16:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/boost/1.63.0-mlhled/include/boost/math/tools/config.hpp:388:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/boost/1.63.0-mlhled/include/boost/detail/fenv.hpp:97:
/cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:58:11: error: no member named 'fenv_t' in the global namespace
  using ::fenv_t;
        ~~^
/cvmfs/cms-ib.cern.ch/nweek-02466/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:59:11: error: no member named 'fexcept_t' in the global namespace
  using ::fexcept_t;
        ~~^
</pre>

--- a/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
+++ b/Geometry/HcalCommonData/src/HcalDDDRecConstants.cc
@@ -597,7 +597,6 @@ void HcalDDDRecConstants::specialRBXHBHE(const std::vector<HcalDetId>& idsOld,
 
 bool HcalDDDRecConstants::specialRBXHBHE(bool tobemerged,
 					 std::vector<HcalDetId>& ids) const {
-  ids.clear();
   if (tobemerged) {
     std::map<HcalDetId,HcalDetId>::const_iterator itr;
     for (itr = detIdSp_.begin(); itr != detIdSp_.end(); ++itr) 

--- a/Geometry/HcalTowerAlgo/test/HcalGeometryPlan1Tester.cc
+++ b/Geometry/HcalTowerAlgo/test/HcalGeometryPlan1Tester.cc
@@ -24,12 +24,12 @@ public:
 
 private:
   edm::ParameterSet ps0_;
-  bool              geomDB_;
+  bool              geomES_;
 };
 
 HcalGeometryPlan1Tester::HcalGeometryPlan1Tester( const edm::ParameterSet& iConfig ) :
   ps0_(iConfig) {
-  geomDB_ = iConfig.getParameter<bool>("GeometryFromDB");
+  geomES_ = iConfig.getParameter<bool>("GeometryFromES");
 }
 
 void HcalGeometryPlan1Tester::analyze(const edm::Event& /*iEvent*/, 
@@ -42,7 +42,7 @@ void HcalGeometryPlan1Tester::analyze(const edm::Event& /*iEvent*/,
   iSetup.get<HcalRecNumberingRecord>().get(topologyHandle);
   const HcalTopology topology = (*topologyHandle);
   HcalGeometry* geom(0);
-  if (geomDB_) {
+  if (geomES_) {
     edm::ESHandle<CaloGeometry> pG;
     iSetup.get<CaloGeometryRecord>().get(pG);
     const CaloGeometry* geo = pG.product();
@@ -57,16 +57,17 @@ void HcalGeometryPlan1Tester::analyze(const edm::Event& /*iEvent*/,
   std::cout << "Special RBX Flag " << ok << " with " << idsp.size()
 	    << " ID's" << std::endl;
   int nall(0), ngood(0);
-  for (unsigned int k=0; k<idsp.size(); ++k) {
-    if (topology.valid(idsp[k])) {
+  for (std::vector<HcalDetId>::const_iterator itr=idsp.begin();
+       itr != idsp.end(); ++itr) {
+    if (topology.valid(*itr)) {
       ++nall;
-      HcalDetId idnew = hcons.mergedDepthDetId(idsp[k]);
-      GlobalPoint pt1 = geom->getGeometry(idsp[k])->getPosition();
+      HcalDetId idnew = hcons.mergedDepthDetId(*itr);
+      GlobalPoint pt1 = geom->getGeometry(*itr)->getPosition();
       GlobalPoint pt2 = geom->getPosition(idnew);
       double     deta = pt1.eta() - pt2.eta();
       double     dphi = pt1.phi() - pt2.phi();
       ok              = (std::abs(deta)<0.00001) && (std::abs(dphi)<0.00001);
-      std::cout << "Unmerged ID " << idsp[k] << " (" << pt1.eta() << ", " 
+      std::cout << "Unmerged ID " << (*itr) << " (" << pt1.eta() << ", " 
 		<< pt1.phi() << ", " << pt1.z() << ") Merged ID " << idnew
 		<< " (" << pt2.eta() << ", " << pt2.phi() << ", " << pt2.z()
 		<< ") ";

--- a/Geometry/HcalTowerAlgo/test/runHcalGeometryPlan1Tester_cfg.py
+++ b/Geometry/HcalTowerAlgo/test/runHcalGeometryPlan1Tester_cfg.py
@@ -12,6 +12,6 @@ process.maxEvents = cms.untracked.PSet(
     )
 
 process.hga = cms.EDAnalyzer("HcalGeometryPlan1Tester",
-                             GeometryFromDB = cms.bool(False))
+                             GeometryFromES = cms.bool(False))
 
 process.p1 = cms.Path(process.hga)

